### PR TITLE
when ringbuffer map size is not a power-of-2 multiple of page size, prompt users in a user-friendly way

### PR DESCRIPTION
**Pull request by hengyoush**
example:
```c
struct {
	__uint(type, BPF_MAP_TYPE_RINGBUF);
	__uint(max_entries, 4097);
} rb SEC(".maps");
```
then output: 
```
program dev_hard_start_xmit: map rb: EINVAL (ring map size 4097 not a power of two page size(pageSize 4096))
```

### CODE REVIEW

File: map.go [hunk 1]
```diff
@@ -432,6 +432,17 @@ func (spec *MapSpec) createMap(inner *sys.FD, opts MapOptions) (_ *Map, err erro
 
 	fd, err := sys.MapCreate(&attr)
 
+	// BPF_MAP_TYPE_RINGBUF's max_entries must be a power-of-2 multiple of kernel's page size.
```
Comment (florianl) [inline map.go:435]: Thanks for your contribution!

Can we move this error check to `handleMapCreateError()`? `handleMapCreateError()` is a central place to extend the returned error with more information.

Comment (hengyoush) [reply map.go]: done

**Review by florianl — APPROVED** [review]

**Comment by lmb** [comment]
I've pushed a small refactoring to reuse an existing implementation of `pow` we already had in the code base.

File: map.go [hunk 2]
```diff
@@ -490,6 +490,16 @@ func handleMapCreateError(attr sys.MapCreateAttr, spec *MapSpec, err error) erro
 			return fmt.Errorf("map create: %w", haveFeatErr)
 		}
 	}
+	// BPF_MAP_TYPE_RINGBUF's max_entries must be a power-of-2 multiple of kernel's page size.
+	if errors.Is(err, unix.EINVAL) &&
+		(attr.MapType == sys.BPF_MAP_TYPE_RINGBUF || attr.MapType == sys.BPF_MAP_TYPE_USER_RINGBUF) {
+		pageSize := uint32(os.Getpagesize())
+		maxEntries := attr.MaxEntries
+		nPages := maxEntries / pageSize
+		if maxEntries%pageSize != 0 || !internal.IsPow(nPages) {
```
Comment (lmb) [inline map.go:499]: @hengyoush I think kernel actually checks that `maxEntries` is a power of two, not `nPages`: https://elixir.bootlin.com/linux/v6.8.4/source/kernel/bpf/ringbuf.c#L194 Do you think that is correct?

**Review by lmb — CHANGES_REQUESTED** [review]

File: map.go (hunk 2 above)
Comment (hengyoush) [reply map.go:499]: Thank you for the correction.

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**